src/storage/txn/process.rs

```diff
@@ -642,7 +642,7 @@ fn process_write_impl<S: Snapshot, L: LockMgr>(
             let key_hashes = gen_key_hashes_if_needed(&lock_mgr, &keys);
 
             let mut txn = MvccTxn::new(snapshot, start_ts, !ctx.get_not_fill_cache())?;
-            let is_pessimistic_txn = txn.cleanup(keys.pop().unwrap(), current_ts)?;
+            let is_pessimistic_txn = txn.cleanup(keys.pop().unwrap(), current_ts, true)?;
```

Comment:
I think you may need to add some comments to explain the reason you passes `true` here, so those who don't know the context can also understand the code.